documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+
+The Authorization Request MUST contain the `scope` parameter, which is a space delimited, case-sensitive list of ASCII strings (scope values). Multiple scope values can be requested simultaneously, subject to the SP being registered to use those “scopes”.
+
+For OpenID requests, the scope values MUST include `openid`, to indicate that the request is an OpenID Connect request, followed by other values depending on the specific CAMARA services being requested.
+
+A purpose must be declared within the requested scope for the following grant types:
+
+* _authorization_code_
+
+* _urn:openid:params:grant-type:ciba_
+
+No purpose is required for the `client_credentials` grant type, as this grant type can only be used when no personal user data is processed. The requested scope must be set directly to the value defined for the relevant endpoint within the OAS ("YAML") specification. "Wild card" scopes (i.e. specifying only the API name) are not valid for this grant type.
+
+To declare a purpose when accessing the CAMARA APIs, it is strongly recommended to guarantee interoperability among Authorization Servers that the `scope` parameter is set to:
+
+`dpv:<dpvValue>#<technicalParameter>`
+
+* `<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+
+* `<technicalParameter>` must be either:
+
+    * one technical scope to limit the request to one API endpoint (this technical scope is described in the API spec YAML file)
+
+    * one API name to cover the complete API (This API name is not described in the API spec YAML file as technical scope) – In this case the request covers all technical scopes of the API.
+
+
+## ID Token
+
+The primary extension that OpenID Connect makes to OAuth 2.0 to enable End-Users to be Authenticated is the ID Token data structure. The ID Token is a security token that contains Claims about the Authentication of an End-User by an Authorization Server when using a Client, and potentially other requested Claims. The ID Token is represented as a [JSON Web Token (JWT)](https://www.rfc-editor.org/info/rfc7519). ID Tokens MUST be signed by Authorization Server using [JWS](https://www.rfc-editor.org/info/rfc7515).
+
+The following Claims are used within the ID Token for all OAuth 2.0 flows used by OpenID Connect:
+
+* **iss**
+
+  REQUIRED. Issuer Identifier for the Issuer of the response (the Authorization Server). The iss value is a case-sensitive URL using the https scheme that contains scheme, host, and optionally, port number and path components and no query or fragment components.
+
+
+* **sub**
+
+  REQUIRED. Subject Identifier. A locally unique and never reassigned identifier within the Issuer for the End-User, which is intended to be consumed by the Client. It MUST NOT exceed 255 ASCII characters in length. The sub value is a case-sensitive string.
```

Comment:
"Globally unique" end user identifiers, such as MSISDN, IMSI or public IP:port, should never be allowed. It should not be possible to identify the end user from the "sub" field alone.

Reply:
That is why it is explained that the sub would be a local unique identifier for authorization server and not reassignable, and the MSISDN or IP could be reassigned.

Reply:
That explanation is not nearly sufficient. A "globally unique" identifier is also "locally unique", and many MSISDNs have never been re-assigned. What must be avoided is that the API provider just echoes the `login_hint` in the `sub` field. The identifier must be pseudonymous, and this needs to be clearly explained.

But your comment raises another issue. If the identifier identifies an end user (i.e. a person) and always identifies that same user, then the API consumer may learn that a given end user is using a new MSISDN if the `sub` field matches a previous `sub` field obtained for a different MSISDN. This is information that they will not necessarily already know, and would count as a privacy breach. So the identifier should not be constant for a given end user for this reason. An end user who changes their MSISDN should get a new `sub`.

Reply:
I have modified the text to indicate that the `sub` claim should be a pseudonymous customer reference.